Python/instrumentation.c

```diff
@@ -2564,114 +2564,99 @@ _PyMonitoring_FireCReturnEvent(PyMonitoringState *state, PyObject *codelike, int
                                      PY_MONITORING_EVENT_C_RETURN);
 }
 
-int
-_PyMonitoring_FirePyThrowEvent(PyMonitoringState *state, PyObject *codelike, int32_t offset,
-                               PyObject *exception)
-{
-    assert(state->active);
+static inline PyObject *
+exception_event_setup(void) {
     PyObject *exc = PyErr_GetRaisedException();
+    assert(exc != NULL);
```

Comment:
Since we're dealing with user code calls here, I wonder if an `assert()` is the right check. I think we should treat this as explicit error and raise an exception to the user. If there is no live exception, it's unlikely that we can do something useful afterwards.

Alternatively, we could return without notification. Not sure how fatal this case really is. If users want us to report an exception event, and there is no exception to report, then which is worse for them: do nothing and be happy, or fail hard and force them to handle the exception? IMHO, if users are really unsure whether to report an exception or not, then `PyErr_Occurred()` isn't far. So an error seems reasonable to me.